pkg/apis/core/v1/defaults.go

```diff
@@ -528,3 +532,14 @@ func defaultHugePagePodLimits(pod *v1.Pod) {
 		pod.Spec.Resources.Limits = podLims
 	}
 }
+
+func defaultPodStatusAllocatedResources(obj *v1.Pod) {
+	if obj.Spec.Resources == nil {
+		return
+	}
+
+	obj.Status.AllocatedResources = make(v1.ResourceList)
```

Comment:
if we keep doing this as generic API defaulting (rather than moving to PrepareForCreate/PrepareForUpdate), this needs to be conditional on obj.Status.AllocatedResources being nil or this isn't defaulting, it's always unconditionally stomping